lib/WarningConfigurationFile.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace SimpleSAML\Module\perun;
+
+use SimpleSAML\Error\Exception;
+use SimpleSAML\Logger;
+use SimpleSAML\Configuration;
+
+/**
+ * Implementation of WarningConfiguration using json file as the source of warning attributes
+ * @package SimpleSAML\Module\perun
+ * @author Dominik Baránek <0Baranek.dominik0@gmail.com>
+ */
+class WarningConfigurationFile
+{
+    const CONFIG_FILE_NAME = 'module_perun.php';
+    const WARNING_FILE = 'disco.warning.file';
+    const WARNING_IS_ON = 'disco.warning.isOn';
+    const WARNING_USER_CAN_CONTINUE = 'disco.warning.userCanContinue';
+    const WARNING_TITLE = 'disco.warning.title';
+    const WARNING_TEXT = 'disco.warning.text';
+
+    public function getSourceOfWarningAttributes()
+    {
+        $file = null;
+        $data = null;
+
+        $config = Configuration::getConfig(self::CONFIG_FILE_NAME);
+
+        try {
+            $file = $config->getString(self::WARNING_FILE);
+
+            set_error_handler(function () {
+                Logger::warning(
+                    "perun:WarningConfigurationFile: " .
+                    "missing or invalid disco.warning.file parameter in module_perun.php"
+                );
+            });
+
+            $json_data = file_get_contents($file);
+            restore_error_handler();
+            $data = json_decode($json_data, true);
+        } catch (\Exception $ex) {
+            Logger::warning(
+                "perun:WarningConfigurationFile: missing or invalid disco.warning.file parameter in module_perun.php"
+            );
+        }
+
+        return $data;
+    }
+
+    public function getWarningAttributes()
+    {
+        $warningIsOn = false;
+        $warningUserCanContinue = true;
+        $warningTitle = '';
+        $warningText = '';
+
+        $data = self::getSourceOfWarningAttributes();
+
+        if (!is_null($data)) {
+            if (isset($data['warningIsOn'])) {
```

Comment:
Please use constant for 'warningIsOn', 'warningUserCanContinue', 'warningTitle', 'warningText'